packages/renderer/src/renderer.js

```diff
@@ -11,6 +11,23 @@ import propsEqual from './utils/propsEqual';
 
 const emptyObject = {};
 
+const appendChild = (parentInstance, child) => {
+  const isParentText = parentInstance.type === 'TEXT';
+  const isChildTextInstance = child.type === 'TEXT_INSTANCE';
+  const isOrphanTextInstance = isChildTextInstance && !isParentText;
+
+  // Ignore orphan text instances.
+  // Caused by cases such as <>{name && <Text>{name}</Text>}</>
+  if (isOrphanTextInstance) {
+    console.warn(
+      `Invalid '${child.value}' string child outside <Text> component`,
+    );
+    return;
```

Comment:
Hi, I think there are some mistakes here. If we haven't push these node, that may got error when `insertBefore` called. 
```
    insertBefore(parentInstance, child, beforeChild) {
      const index = parentInstance.children?.indexOf(beforeChild);

      if (index === undefined) return;

      if (index !== -1 && child)
        parentInstance.children.splice(index, 0, child);
    },
```
child will be droped if beforeChild is not OrphanTextInstance

Reply:
Could you provide a test that shows the error?

Reply:
```
import { Document, Font, Image, Link, Page, PDFViewer, StyleSheet, Text, View } from '@react-pdf/renderer'
import ReactMarkdown from 'react-markdown'
import { ReactMarkdownProps } from 'react-markdown/lib/ast-to-react'

const App = () => (
  <PDFViewer style={{ width: '100%', height: 800 }}>
    <Document>
      <Page style={styles.body}>
        {/*<Text>*/}
        <ReactMarkdown components={{
          h1: ({ children }: ReactMarkdownProps) => <Text>{children}</Text>,
          p: ({ children }: ReactMarkdownProps) => <Text>{children}</Text>,
        }}>{`# a
b`}</ReactMarkdown>
        {/*</Text>*/}
      </Page>
    </Document>
  </PDFViewer>
)
```

![image](https://user-images.githubusercontent.com/3294100/230276495-396971ef-e0b2-48ea-a699-fa08dcda548c.png)

if add `<Text>` out of `ReactMarkdown`:

![image](https://user-images.githubusercontent.com/3294100/230276759-cea2fb28-031d-4eab-a748-7f3dadcfbb48.png)

Actually, That can be found without ReactMarkdown. But I don't know when will insertBefore be called.